Aprox.cpp

```diff
@@ -0,0 +1,39 @@
+#include "Aprox.h"
+
+Aprox::Aprox(tsp::Graph<double, std::vector> &graph_) : Algorithm(graph_) {
+  std::cout << "New Aprox constructed" << std::endl;
+}
+
+Aprox::~Aprox() {
+  std::cout << "2-approximate solution:\n";
+  for (const auto vertex : solution) {
+    std::cout << vertex << " ";
+  }
+  std::cout << std::endl;
+  std::cout << "Aprox deconstructed" << std::endl;
+}
+
+void Aprox::calculate_solution() {
+  const auto N = graph.get_order();
+  const auto &G = graph.get_graph();
+
+  std::vector<bool> vertex_included_in_solution(N, false);
+  for (unsigned i = 0; i < N; ++i) {
+    if (!vertex_included_in_solution[i]) {
+      for (unsigned j = 0; j < N; ++j) {
```

Comment:
1. why casting from int to unsigned? use unsigned/auto j = 0u;
2. why not j = i + 1?

Reply:
1. No casting involved, but gonna change to auto.
2. Good catch.